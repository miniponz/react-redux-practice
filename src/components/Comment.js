import React from 'react';
import PropTypes from 'prop-types';

function Comment({ id, comment, deleteComment }){
  return (
    <div>
      <p>{comment}</p>
      <button onClick={deleteComment.bind(null, id)}>...delete comment?</button>
    </div>
  );
}

Comment.propTypes = {
  comment: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  deleteComment: PropTypes.func.isRequired
};

export default Comment;

