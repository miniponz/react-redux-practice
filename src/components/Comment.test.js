import React from 'react';
import { shallow } from 'enzyme';
import Comment from './Comment';

describe('Display Component', () => {
  it('displays stuff!', () => {
    const wrapper = shallow(<Comment deleteComment={jest.fn()} comment={'text'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
