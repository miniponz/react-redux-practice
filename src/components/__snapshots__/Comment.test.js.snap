// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Component displays stuff! 1`] = `
<div>
  <p>
    text
  </p>
  <button
    onClick={[Function]}
  >
    ...delete comment?
  </button>
</div>
`;
